eip_operator/src/eip.rs

```diff
@@ -335,26 +385,27 @@ pub(crate) async fn set_status_created(
 #[instrument(skip(api), err)]
 pub(crate) async fn set_status_attached(
     api: &Api<Eip>,
-    name: &str,
+    eip: &Eip,
     eni: &str,
     private_ip_address: &str,
-) -> Result<Eip, kube::Error> {
+    resource_id: &str,
+) -> Result<Eip, Error> {
     event!(Level::INFO, "Updating status for attached EIP.");
-    let patch = serde_json::json!({
-        "apiVersion": Eip::version(),
-        "kind": "Eip",
-        "status": {
-            "eni": eni,
-            "privateIpAddress": private_ip_address,
-        }
-    });
-    let patch = Patch::Merge(&patch);
-    let params = PatchParams::default();
-    let result = api.patch_status(name, &params, &patch).await;
-    if result.is_ok() {
-        event!(Level::INFO, "Done updating status for attached EIP.");
-    }
-    result
+    let mut eip = eip.clone();
+    let status = eip.status.as_mut().ok_or(Error::MissingEipStatus)?;
+    status.eni = Some(eni.to_owned());
+    status.private_ip_address = Some(private_ip_address.to_owned());
+    status.resource_id = Some(resource_id.to_owned());
```

Comment:
As commented above, perhaps this should be moved to its own `set_status_resource_id` function.